apps/api/v2/src/ee/event-types-private-links/controllers/organizations-event-types-private-links.controller.ts

```diff
@@ -0,0 +1,137 @@
+import { API_VERSIONS_VALUES } from "@/lib/api-versions";
+import {
+  OPTIONAL_API_KEY_HEADER,
+  OPTIONAL_X_CAL_CLIENT_ID_HEADER,
+  OPTIONAL_X_CAL_SECRET_KEY_HEADER,
+} from "@/lib/docs/headers";
+import { PlatformPlan } from "@/modules/auth/decorators/billing/platform-plan.decorator";
+import { GetUser } from "@/modules/auth/decorators/get-user/get-user.decorator";
+import { Roles } from "@/modules/auth/decorators/roles/roles.decorator";
+import { ApiAuthGuard } from "@/modules/auth/guards/api-auth/api-auth.guard";
+import { PlatformPlanGuard } from "@/modules/auth/guards/billing/platform-plan.guard";
+import { IsAdminAPIEnabledGuard } from "@/modules/auth/guards/organizations/is-admin-api-enabled.guard";
+import { IsOrgGuard } from "@/modules/auth/guards/organizations/is-org.guard";
+import { RolesGuard } from "@/modules/auth/guards/roles/roles.guard";
+import { IsTeamInOrg } from "@/modules/auth/guards/teams/is-team-in-org.guard";
+import { TeamsEventTypesService } from "@/modules/teams/event-types/services/teams-event-types.service";
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  ParseIntPipe,
+  Patch,
+  Post,
+  UseGuards,
+} from "@nestjs/common";
+import { ApiHeader, ApiOperation, ApiTags as DocsTags } from "@nestjs/swagger";
+
+import { SUCCESS_STATUS } from "@calcom/platform-constants";
+import {
+  CreatePrivateLinkInput,
+  CreatePrivateLinkOutput,
+  DeletePrivateLinkOutput,
+  GetPrivateLinksOutput,
+  UpdatePrivateLinkInput,
+  UpdatePrivateLinkOutput,
+} from "@calcom/platform-types";
+
+import { PrivateLinksService } from "../services/private-links.service";
+
+@Controller({
+  path: "/v2/organizations/:orgId/teams/:teamId/event-types/:eventTypeId/private-links",
+  version: API_VERSIONS_VALUES,
+})
+@DocsTags("Orgs / Teams / Event Types / Private Links")
+@ApiHeader(OPTIONAL_X_CAL_CLIENT_ID_HEADER)
+@ApiHeader(OPTIONAL_X_CAL_SECRET_KEY_HEADER)
+@ApiHeader(OPTIONAL_API_KEY_HEADER)
+export class OrganizationsEventTypesPrivateLinksController {
+  constructor(
+    private readonly privateLinksService: PrivateLinksService,
+    private readonly teamsEventTypesService: TeamsEventTypesService
+  ) {}
+
+  @Post("/")
+  @Roles("TEAM_ADMIN")
+  @PlatformPlan("ESSENTIALS")
+  @UseGuards(ApiAuthGuard, IsOrgGuard, RolesGuard, IsTeamInOrg, PlatformPlanGuard, IsAdminAPIEnabledGuard)
+  @ApiOperation({ summary: "Create a private link for a team event type" })
+  async createPrivateLink(
+    @Param("teamId", ParseIntPipe) teamId: number,
+    @Param("eventTypeId", ParseIntPipe) eventTypeId: number,
+    @Body() body: CreatePrivateLinkInput,
+    @GetUser("id") userId: number
+  ): Promise<CreatePrivateLinkOutput> {
+    await this.teamsEventTypesService.validateEventTypeExists(teamId, eventTypeId);
+    const privateLink = await this.privateLinksService.createPrivateLink(eventTypeId, userId, body);
```

Comment:
Question - this pr is for org team private links but it creates private links in the same way as for individual user
```
const privateLink = await this.privateLinksService.createPrivateLink(eventTypeId, userId, body);
```

so what is the difference? Why are private links not associated with a teamId but userId ? Because authenticated user is team admin and his / her userId is passed now in the new endpoint but the same would happen in individual endpoint. The same question for rest of endpoints.

Reply:
Diff:
- Individual endpoints: access controlled by EventTypeOwnershipGuard (user must own the event type).
- Org/team endpoints: access controlled by org/team guards (IsOrgGuard, IsTeamInOrg, RolesGuard TEAM_ADMIN, PlatformPlanGuard, IsAdminAPIEnabledGuard) and we validate the event type belongs to the team. This lets org/team admins manage links for team-owned event types.

The userId you see in the call isn’t persisted; it’s used to generate the link string (generateHashedLink) , not for authorization. It was unnecessarily being passed through the GET, PATCH and DELETE, which I've removed if from now. Authorization is handled by the guards and by checking the event type’s ownership/scope.